arcgis-ios-sdk-samples/Route & Directions/Find closest facility to an incident (interactive)/FindClosestFacilityInteractiveViewController.swift

```diff
@@ -0,0 +1,125 @@
+//
+// Copyright © 2020 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import UIKit
+import ArcGIS
+
+class FindClosestFacilityInteractiveViewController: UIViewController {
+    @IBOutlet var mapView: AGSMapView! {
+        didSet {
+            // Initialize the map.
+            mapView.map = AGSMap(basemapType: .streets, latitude: 32.727, longitude: -117.1750, levelOfDetail: 12)
+            mapView.touchDelegate = self
+            
+            // Create symbols and graphics to add to the graphic overlays.
+            createFacilitiesAndGraphics()
+            mapView.graphicsOverlays.add(facilityGraphicsOverlay)
+            mapView.graphicsOverlays.add(incidentGraphicsOverlay)
+        }
+    }
+    
+    private let facilityURL = URL(string: "https://static.arcgis.com/images/Symbols/SafetyHealth/Hospital.png")!
+    
+    // Create a closest facility task from the network service URL.
+    private let closestFacilityTask: AGSClosestFacilityTask = {
+        let networkServiceURL = URL(string: "https://sampleserver6.arcgisonline.com/arcgis/rest/services/NetworkAnalysis/SanDiego/NAServer/ClosestFacility")!
+        return AGSClosestFacilityTask(url: networkServiceURL)
+    }()
+    
+    // Create graphic overlays to the map.
+    private var facilityGraphicsOverlay = AGSGraphicsOverlay()
+    private var incidentGraphicsOverlay = AGSGraphicsOverlay()
+    
+    // Create graphics to represent the route.
+    private let routeSymbol = AGSSimpleLineSymbol(style: .solid, color: .blue, width: 2.0)
+    
+    // Create an array of facilities in the area.
+    private var facilities = [AGSFacility]()
+    
+    // Add the facilities and create graphics.
+    private func createFacilitiesAndGraphics() {
```

Comment:
By starting with a verb like _create_, I think this method is going to return something. Here are a few ideas:

- Move creation of facilities to the property itself:

  ```swift
  // Create an array of facilities in the area.
  private let facilities = [
      AGSFacility(point: AGSPoint(x: -1.3042129900625112E7, y: 3860127.9479775648, spatialReference: .webMercator())),
      AGSFacility(point: AGSPoint(x: -1.3042193400557665E7, y: 3862448.873041752, spatialReference: .webMercator())),
      AGSFacility(point: AGSPoint(x: -1.3046882875518233E7, y: 3862704.9896770366, spatialReference: .webMercator())),
      AGSFacility(point: AGSPoint(x: -1.3040539754780494E7, y: 3862924.5938606677, spatialReference: .webMercator())),
      AGSFacility(point: AGSPoint(x: -1.3042571225655518E7, y: 3858981.773018156, spatialReference: .webMercator())),
      AGSFacility(point: AGSPoint(x: -1.3039784633928463E7, y: 3856692.5980474586, spatialReference: .webMercator())),
      AGSFacility(point: AGSPoint(x: -1.3049023883956768E7, y: 3861993.789732541, spatialReference: .webMercator()))
  ]
  ```

- Repurpose the method to create the facilities overlay itself since the `facilityGraphicsOverlay` property isn't used after the overlay is added to the map view:

  ```swift
  /// Creates a graphics overlay with graphics for each of the facilities.
  func makeFacilitiesOverlay() -> AGSGraphicsOverlay {
      let facilitySymbol = AGSPictureMarkerSymbol(url: facilityURL)
      facilitySymbol.height = 30
      facilitySymbol.width = 30
  
      let graphicsOverlay = AGSGraphicsOverlay()
      graphicsOverlay.graphics.addObjects(from: facilities.map {
          AGSGraphic(geometry: $0.geometry, symbol: facilitySymbol)
      })
      return graphicsOverlay
  }
   ```

- Add the facilities overlay to the map view without assigning it to a property:

  ```swift
  mapView.graphicsOverlays.add(makeFacilitiesOverlay())
  ```